src/v2/trades.ts

```diff
@@ -0,0 +1,63 @@
+import { getAddress } from '@ethersproject/address'
+import { NowRequest, NowResponse } from '@now/node'
+import { BigNumber } from '@uniswap/sdk'
+
+import { getSwaps } from './_shared'
+import { return200, return400, return500 } from '../utils'
+
+export default async function(req: NowRequest, res: NowResponse): Promise<void> {
+  if (
+    !req.query.pair ||
+    typeof req.query.pair !== 'string' ||
+    !/^0x[0-9a-fA-F]{40}_0x[0-9a-fA-F]{40}$/.test(req.query.pair)
+  ) {
+    return400(res, 'Invalid pair identifier: must be of format tokenAddress_tokenAddress')
+    return
+  }
+
+  const [tokenA, tokenB] = req.query.pair.split('_')
+  let idA: string, idB: string
+  try {
+    ;[idA, idB] = [getAddress(tokenA), getAddress(tokenB)]
+  } catch (error) {
+    return400(res)
+    return
+  }
```

Comment:
might be worth extracting into a helper?

Reply:
thought about it, checked the v1 endpoints, figured it's not worth it